firebase_admin/mlkit.py

```diff
@@ -459,6 +550,85 @@ def __init__(self, app):
         self._client = _http_client.JsonHttpClient(
             credential=app.credential.get_credential(),
             base_url=self._project_url)
+        self._operation_client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_MLKitService.OPERATION_URL)
+
+    def get_operation(self, op_name):
+        _validate_and_parse_operation_name(op_name)
+        try:
+            return self._operation_client.body('get', url=op_name)
+        except requests.exceptions.RequestException as error:
+            raise _utils.handle_platform_error_from_requests(error)
+
+    def handle_operation(self, operation, max_polling_attempts=None, max_time_seconds=None,
+                         always_return_model=False):
+        """Handles long running operations.
+
+        Args:
+            operation: The operation to handle.
+            max_polling_attempts: The maximum number of polling requests to make.
+                (None for no limit)
+            max_time_seconds: The maximum seconds to try polling for operation complete.
+                (None for no limit)
+            always_return_model: If true, returns a locked Model instead of raising deadline
+                exceeded exceptions.
+
+        Returns:
+            dict: A dictionary of the returned model properties.
+
+        Raises:
+            TypeError: if the operation is not a dictionary.
+            ValueError: If the operation is malformed.
+        """
+        if not isinstance(operation, dict):
+            raise TypeError('Operation must be a dictionary.')
+        op_name = operation.get('name')
+        _, model_id = _validate_and_parse_operation_name(op_name)
+
+        current_attempt = 0
+        start_time = datetime.datetime.now()
+        stop_time = (None if max_time_seconds is None else
+                     start_time + datetime.timedelta(seconds=max_time_seconds))
+        while True:
+            if operation.get('done'):
+                if operation.get('response'):
+                    return operation.get('response')
+                elif operation.get('error'):
+                    raise _utils.handle_operation_error(operation.get('error'))
+                else:
+                    # A 'done' operation must have either a response or an error.
+                    raise ValueError('Operation is malformed.')
+            else:
+                # We just got this operation. Wait before getting another
+                # so we don't exceed the GetOperation maximum request rate.
+                if max_polling_attempts is not None and current_attempt >= max_polling_attempts:
+                    if always_return_model:
+                        return get_model(model_id).as_dict()
+                    raise exceptions.DeadlineExceededError('Polling max attempts exceeded.')
+                delay_factor = pow(
```

Comment:
Extract this part (up until sleep) into a helper method.

```
def _exponential_backoff(self, current_attempt, stop_time):
    # calculate delay, validate and sleep
```

Reply:
done